tests/undefined_symbol_in_sdfg_test.py

```diff
@@ -0,0 +1,344 @@
+    read_tasklet = state.add_tasklet('read_tmp', {'t'}, {'b'}, 'b = t')
+
+    # Connect everything
+    state.add_edge(A, None, outer_map_entry, None, dace.Memlet())
+    state.add_edge(outer_map_entry, None, inner_map_entry, None, dace.Memlet())
+    state.add_edge(inner_map_entry, None, tasklet, None, dace.Memlet())
+    state.add_edge(A, None, tasklet, 'a', dace.Memlet('A[i, j]'))
+    state.add_edge(tasklet, 't', tmp, None, dace.Memlet('tmp[i, 0]'))
+    state.add_edge(inner_map_entry, None, read_tasklet, None, dace.Memlet())
+    state.add_edge(tmp, None, read_tasklet, 't', dace.Memlet('tmp[i, 0]'))
+    state.add_edge(read_tasklet, 'b', B, None, dace.Memlet('B[i]'))
+    state.add_edge(read_tasklet, None, inner_map_exit, None, dace.Memlet())
+    state.add_edge(inner_map_exit, None, outer_map_exit, None, dace.Memlet())
+
+    # Try to generate code - this should fail because of the undefined symbol
+    with pytest.raises((TypeError, dace.sdfg.InvalidSDFGError), match=r"undefined symbol|contains undefined symbol"):
+        sdfg.compile()
+
+
+def test_undefined_symbol_in_unused_dimension():
+    """Tests that validation catches UndefinedSymbol even in 'unused' dimensions."""
+
+    # Create an SDFG manually
+    sdfg = dace.SDFG('undefined_symbol_unused_test')
+
+    # Define symbols
+    undefined_dim = symbolic.UndefinedSymbol()
+
+    # Add arrays to SDFG - in this case the undefined dimension is in the array A
+    # which is an input (non-transient) so it doesn't need to be allocated
+    sdfg.add_array('A', [undefined_dim, 20], dace.float64)
+    sdfg.add_array('B', [20], dace.float64)
+    sdfg.add_transient('tmp', [20, 20], dace.float64)
+
+    # Create state
+    state = sdfg.add_state()
+
+    # Create access nodes
+    A = state.add_read('A')
+    tmp = state.add_access('tmp')
+    B = state.add_write('B')
+
+    # Add map
+    map_entry, map_exit = state.add_map('compute', dict(i='0:20'))
+
+    # Add tasklets
+    tasklet_write = state.add_tasklet('write', {'a'}, {'t'}, 't = a')
+    tasklet_read = state.add_tasklet('read', {'t'}, {'b'}, 'b = t')
+
+    # Connect everything - only using the defined dimension (index 1),
+    # avoiding the undefined one (index 0)
+    state.add_edge(A, None, map_entry, None, dace.Memlet())
+    state.add_edge(map_entry, None, tasklet_write, None, dace.Memlet())
+    state.add_edge(A, None, tasklet_write, 'a', dace.Memlet('A[0, i]'))
+    state.add_edge(tasklet_write, 't', tmp, None, dace.Memlet('tmp[i, 0]'))
+    state.add_edge(tmp, None, tasklet_read, 't', dace.Memlet('tmp[i, 0]'))
+    state.add_edge(tasklet_read, 'b', B, None, dace.Memlet('B[i]'))
+    state.add_edge(tasklet_read, None, map_exit, None, dace.Memlet())
+
+    # Even though we only access A[0, i] (a constant in the first dimension),
+    # validation should still fail because the UndefinedSymbol appears in the argument
+    with pytest.raises(InvalidSDFGError, match="undefined symbol in dimension"):
+        sdfg.validate()
```

Comment:
This is incorrect. The above program is a legitimate use case of undefined symbols. In fact, the undefined symbol should _not_ appear in the argument list. I am sorry to have caused confusion, but this test should succeed, and thus you should test both sdfg.validate() and sdfg.compile().